Unit test fails for networkAnalyzer due to getRecord()

**Describe the bug**
The unit test fails for networkAnalyzer due to the method getRecord().
```
=== RUN   TestHttpProtocol
=== RUN   TestHttpProtocol/slowData
=== RUN   TestHttpProtocol/errorData
=== RUN   TestHttpProtocol/normal
--- PASS: TestHttpProtocol (0.01s)
    --- PASS: TestHttpProtocol/slowData (0.00s)
    --- PASS: TestHttpProtocol/errorData (0.00s)
    --- PASS: TestHttpProtocol/normal (0.00s)
=== RUN   TestMySqlProtocol
=== RUN   TestMySqlProtocol/query-split
=== RUN   TestMySqlProtocol/query
--- PASS: TestMySqlProtocol (0.00s)
    --- PASS: TestMySqlProtocol/query-split (0.00s)
    --- PASS: TestMySqlProtocol/query (0.00s)
=== RUN   TestRedisProtocol
=== RUN   TestRedisProtocol/get
--- PASS: TestRedisProtocol (0.00s)
    --- PASS: TestRedisProtocol/get (0.00s)
=== RUN   TestDnsProtocol
=== RUN   TestDnsProtocol/multi
    network_analyzer_test.go:258: [Check request_sent_time] want=5000, got=405000
    network_analyzer_test.go:258: [Check waiting_ttfb_time] want=970000, got=570000
    network_analyzer_test.go:258: [Check content_download_time] want=30000, got=530000
    network_analyzer_test.go:258: [Check request_total_time] want=1005000, got=1505000
    network_analyzer_test.go:258: [Check request_io] want=42, got=84
    network_analyzer_test.go:258: [Check response_io] want=89, got=162
    network_analyzer_test.go:258: [Check request_sent_time] want=4000, got=405000
    network_analyzer_test.go:258: [Check waiting_ttfb_time] want=1020000, got=570000
    network_analyzer_test.go:258: [Check content_download_time] want=80000, got=530000
    network_analyzer_test.go:258: [Check request_total_time] want=1104000, got=1505000
    network_analyzer_test.go:258: [Check request_io] want=42, got=84
    network_analyzer_test.go:258: [Check response_io] want=73, got=162
--- FAIL: TestDnsProtocol (0.00s)
    --- FAIL: TestDnsProtocol/multi (0.00s)
=== RUN   TestKafkaProtocol
=== RUN   TestKafkaProtocol/produce-split
=== RUN   TestKafkaProtocol/fetch-split
--- PASS: TestKafkaProtocol (0.00s)
    --- PASS: TestKafkaProtocol/produce-split (0.00s)
    --- PASS: TestKafkaProtocol/fetch-split (0.00s)
FAIL
exit status 1
FAIL    github.com/Kindling-project/kindling/collector/analyzer/network 0.037s
```

**How to reproduce?**
Just run `go test -v` under the directory of `collector/analyzer/network/`.

**Environment (please complete the following information)**
Based on the latest commit 4e6c9d9f755def545703aa0d9e85197de189c0a8.

**What causes this issue**
This is caused by typos in the function `getRecord()`. This function is used to get the final `GaugeGroup` from a `messagePair` instead of `messagePairs`. It is different from the function `getRecords()` which does that from `messagePairs`. The difference exists because `getRecord()` is used for the protocols, like DNS, which could send multiple real requests before receiving the responses, and `getRercords()` is used for protocols, like HTTP, which can only send one real request before receiving the response. See the following diagrams for details.

The protocols, like DNS, could send multiple real requests before receiving the responses.
<img width="605" alt="image" src="https://user-images.githubusercontent.com/46807570/161187583-9f840738-9824-4471-8d1d-6ea030980b3c.png">

While the protocols, like HTTP, only send one real request before receiving a response.  
<img width="585" alt="image" src="https://user-images.githubusercontent.com/46807570/161187597-e7536efc-a394-4fc0-855f-2ef60b7e505e.png">

But for our data structure `messagePairs`, there could be multiple request events and response events in one `messagePairs`. For HTTP, it means one real request has multiple parts, while for DNS, it means multiple real requests.
<img width="653" alt="image" src="https://user-images.githubusercontent.com/46807570/161187616-e35decdb-2373-4eee-aba3-f1cc915822f0.png">

Therefore, the difference between DNS and HTTP brings different parsing methods, `getRecord` and `getRecords`. This issue is caused by the misunderstanding of what `getRecord` does.

--- a/collector/analyzer/network/network_analyzer.go
+++ b/collector/analyzer/network/network_analyzer.go
@@ -368,69 +368,8 @@ func (na *NetworkAnalyzer) parseProtocols(mps *messagePairs) []*model.GaugeGroup
 
 func (na *NetworkAnalyzer) parseProtocol(mps *messagePairs, parser *protocol.ProtocolParser) []*model.GaugeGroup {
 	if parser.MultiRequests() {
-		// Match with key when disordering.
-		size := mps.requests.size()
-		parsedReqMsgs := make([]*protocol.PayloadMessage, size)
-		for i := 0; i < size; i++ {
-			req := mps.requests.getEvent(i)
-			requestMsg := protocol.NewRequestMessage(req.GetData())
-			if !parser.ParseRequest(requestMsg) {
-				// Parse failure
-				return nil
-			}
-			parsedReqMsgs[i] = requestMsg
-		}
-
-		records := make([]*model.GaugeGroup, 0)
-		if mps.responses == nil {
-			size := mps.requests.size()
-			for i := 0; i < size; i++ {
-				req := mps.requests.getEvent(i)
-				mp := &messagePair{
-					request:  req,
-					response: nil,
-				}
-				records = append(records, na.getRecord(mps, mp, parser.GetProtocol(), parsedReqMsgs[i].GetAttributes()))
-			}
-			return records
-		} else {
-			matchedRequestIdx := make(map[int]bool)
-			size := mps.responses.size()
-			for i := 0; i < size; i++ {
-				resp := mps.responses.getEvent(i)
-				responseMsg := protocol.NewResponseMessage(resp.GetData(), model.NewAttributeMap())
-				if !parser.ParseResponse(responseMsg) {
-					// Parse failure
-					return nil
-				}
-				// Match Request with repsone
-				matchIdx := parser.PairMatch(parsedReqMsgs, responseMsg)
-				if matchIdx == -1 {
-					return nil
-				}
-				matchedRequestIdx[matchIdx] = true
-
-				mp := &messagePair{
-					request:  mps.requests.getEvent(matchIdx),
-					response: resp,
-				}
-				records = append(records, na.getRecord(mps, mp, parser.GetProtocol(), responseMsg.GetAttributes()))
-			}
-			// 498 Case
-			reqSize := mps.requests.size()
-			for i := 0; i < reqSize; i++ {
-				req := mps.requests.getEvent(i)
-				if _, matched := matchedRequestIdx[i]; !matched {
-					mp := &messagePair{
-						request:  req,
-						response: nil,
-					}
-					records = append(records, na.getRecord(mps, mp, parser.GetProtocol(), parsedReqMsgs[i].GetAttributes()))
-				}
-			}
-
-			return records
-		}
+		// Not mergable requests
+		return na.parseMultipleRequests(mps, parser)
 	}
 
 	// Mergable Data
@@ -450,6 +389,73 @@ func (na *NetworkAnalyzer) parseProtocol(mps *messagePairs, parser *protocol.Pro
 	}
 
 	return na.getRecords(mps, parser.GetProtocol(), responseMsg.GetAttributes())
+}
+
+// parseMultipleRequests parses the messagePairs when we know there could be multiple read requests.
+// This is used only when the protocol is DNS now.
+func (na *NetworkAnalyzer) parseMultipleRequests(mps *messagePairs, parser *protocol.ProtocolParser) []*model.GaugeGroup {
+	// Match with key when disordering.
+	size := mps.requests.size()
+	parsedReqMsgs := make([]*protocol.PayloadMessage, size)
+	for i := 0; i < size; i++ {
+		req := mps.requests.getEvent(i)
+		requestMsg := protocol.NewRequestMessage(req.GetData())
+		if !parser.ParseRequest(requestMsg) {
+			// Parse failure
+			return nil
+		}
+		parsedReqMsgs[i] = requestMsg
+	}
+
+	records := make([]*model.GaugeGroup, 0)
+	if mps.responses == nil {
+		size := mps.requests.size()
+		for i := 0; i < size; i++ {
+			req := mps.requests.getEvent(i)
+			mp := &messagePair{
+				request:  req,
+				response: nil,
+			}
+			records = append(records, na.getRecordWithSinglePair(mps, mp, parser.GetProtocol(), parsedReqMsgs[i].GetAttributes()))
+		}
+		return records
+	} else {
+		matchedRequestIdx := make(map[int]bool)
+		size := mps.responses.size()
+		for i := 0; i < size; i++ {
+			resp := mps.responses.getEvent(i)
+			responseMsg := protocol.NewResponseMessage(resp.GetData(), model.NewAttributeMap())
+			if !parser.ParseResponse(responseMsg) {
+				// Parse failure
+				return nil
+			}
+			// Match Request with repsone
+			matchIdx := parser.PairMatch(parsedReqMsgs, responseMsg)
+			if matchIdx == -1 {
+				return nil
+			}
+			matchedRequestIdx[matchIdx] = true
+
+			mp := &messagePair{
+				request:  mps.requests.getEvent(matchIdx),
+				response: resp,
+			}
+			records = append(records, na.getRecordWithSinglePair(mps, mp, parser.GetProtocol(), responseMsg.GetAttributes()))
+		}
+		// 498 Case
+		reqSize := mps.requests.size()
+		for i := 0; i < reqSize; i++ {
+			req := mps.requests.getEvent(i)
+			if _, matched := matchedRequestIdx[i]; !matched {
+				mp := &messagePair{
+					request:  req,
+					response: nil,
+				}
+				records = append(records, na.getRecordWithSinglePair(mps, mp, parser.GetProtocol(), parsedReqMsgs[i].GetAttributes()))
+			}
+		}
+		return records
+	}
 }
 
 func (na *NetworkAnalyzer) getConnectFailRecords(mps *messagePairs) []*model.GaugeGroup {
@@ -522,7 +528,10 @@ func (na *NetworkAnalyzer) getRecords(mps *messagePairs, protocol string, attrib
 	return []*model.GaugeGroup{ret}
 }
 
-func (na *NetworkAnalyzer) getRecord(mps *messagePairs, mp *messagePair, protocol string, attributes *model.AttributeMap) *model.GaugeGroup {
+// getRecordWithSinglePair generates a record whose metrics are copied from the input messagePair,
+// instead of messagePairs. This is used only when there could be multiple real requests in messagePairs.
+// For now, only messagePairs with DNS protocol would run into this method.
+func (na *NetworkAnalyzer) getRecordWithSinglePair(mps *messagePairs, mp *messagePair, protocol string, attributes *model.AttributeMap) *model.GaugeGroup {
 	evt := mp.request
 
 	slow := na.isSlow(mp.getDuration(), protocol)
@@ -555,12 +564,12 @@ func (na *NetworkAnalyzer) getRecord(mps *messagePairs, mp *messagePair, protoco
 	}
 
 	ret.UpdateAddGauge(constvalues.ConnectTime, 0)
-	ret.UpdateAddGauge(constvalues.RequestSentTime, mps.getSentTime())
-	ret.UpdateAddGauge(constvalues.WaitingTtfbTime, mps.getWaitingTime())
-	ret.UpdateAddGauge(constvalues.ContentDownloadTime, mps.getDownloadTime())
-	ret.UpdateAddGauge(constvalues.RequestTotalTime, int64(mps.getDuration()))
-	ret.UpdateAddGauge(constvalues.RequestIo, int64(mps.getRquestSize()))
-	ret.UpdateAddGauge(constvalues.ResponseIo, int64(mps.getResponseSize()))
+	ret.UpdateAddGauge(constvalues.RequestSentTime, mp.getSentTime())
+	ret.UpdateAddGauge(constvalues.WaitingTtfbTime, mp.getWaitingTime())
+	ret.UpdateAddGauge(constvalues.ContentDownloadTime, mp.getDownloadTime())
+	ret.UpdateAddGauge(constvalues.RequestTotalTime, int64(mp.getDuration()))
+	ret.UpdateAddGauge(constvalues.RequestIo, int64(mp.getRquestSize()))
+	ret.UpdateAddGauge(constvalues.ResponseIo, int64(mp.getResponseSize()))
 
 	ret.Timestamp = evt.GetStartTime()
 	return ret

